tests/ga/test_extension.py

```diff
@@ -79,103 +81,128 @@ def raise_ioerror(*args):
     e.errno = EIO
     raise e
 
-
 class TestExtensionCleanup(AgentTestCase):
+
     def setUp(self):
         AgentTestCase.setUp(self)
-        self.ext_handlers = ExtHandlersHandler(Mock())
-        self.lib_dir = tempfile.mkdtemp()
-
-    def _install_handlers(self, start=0, count=1,
-                          handler_state=ExtHandlerState.Installed):
-        src = os.path.join(data_dir, "ext", "sample_ext-1.3.0.zip")
-        version = FlexibleVersion("1.3.0")
-        version += start - version.patch
-
-        for i in range(start, start + count):
-            eh = ExtHandler()
-            eh.name = "sample_ext"
-            eh.properties.version = str(version)
-            handler = ExtHandlerInstance(eh, "unused")
-
-            dst = os.path.join(self.lib_dir,
-                               handler.get_full_name() + HANDLER_PKG_EXT)
-            shutil.copy(src, dst)
-
-            if not handler_state is None:
-                zipfile.ZipFile(dst).extractall(handler.get_base_dir())
-                handler.set_handler_state(handler_state)
+        self.mock_sleep = patch("time.sleep", lambda *_: mock_sleep(0.0001))
+        self.mock_sleep.start()
 
-            version += 1
+    def tearDown(self):
+        AgentTestCase.tearDown(self)
```

Comment:
the order of cleanup should be the reverse of setup (i.e. stop the mock first, then invoke the base class)

Reply:
Didnt, know that, thanks for letting me know! Done

Reply:
to clarify - it is just a best practice; in this case it does not make a difference